From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: System Builder <builder@nerves-project.org>
Date: Wed, 30 Jul 2025 10:00:00 +0000
Subject: [PATCH] Fix CNSS conditional compilation for open source builds

The CNSS conditional compilation is not working correctly and is trying
to call non-existent cnss_* functions. This patch forces the open source
path by ensuring the correct conditional is taken.

In open source builds, we should use the static inline implementations
provided in the WLAN_OPEN_SOURCE && !CONFIG_CNSS section, not the
proprietary cnss_* function calls.

Signed-off-by: System Builder <builder@nerves-project.org>
---
 CORE/VOSS/inc/vos_cnss.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CORE/VOSS/inc/vos_cnss.h b/CORE/VOSS/inc/vos_cnss.h
index 1234567..abcdefg 100644
--- a/CORE/VOSS/inc/vos_cnss.h
+++ b/CORE/VOSS/inc/vos_cnss.h
@@ -32,7 +32,7 @@
 
 #define DISABLE_KRAIT_IDLE_PS_VAL    1
 
-#if defined(WLAN_OPEN_SOURCE) && !defined(CONFIG_CNSS)
+#if 1 /* Force open source path for kernel 5.4 compatibility */
 #include <linux/device.h>
 #include <linux/pm_wakeup.h>
 #include <linux/jiffies.h>
-- 
2.39.2
